src/components/Layout/Layout.tsx

```diff
@@ -26,32 +26,49 @@ import { LEFT_NAV_LINKS, RIGHT_NAV_LINKS } from '@/utils/globalnav';
 import { trackPageVisit } from '../../utils/track';
 import { Menu } from '@/components/Menu';
 import { LayoutProvider } from '@/components/Layout';
+import { TableOfContents } from '@/components/TableOfContents';
+import type { Heading } from '@/components/TableOfContents/TableOfContents';
 import directory from 'src/directory/directory.json';
 import { PageNode } from 'src/directory/directory';
 
-export const Layout = forwardRef(function Layout(
-  {
-    children,
-    pageTitle,
-    pageDescription,
-    platform,
-    url,
-    pageType = 'inner'
-  }: {
-    children: any;
-    pageTitle?: string;
-    pageDescription?: string;
-    platform?: Platform;
-    url?: string;
-    pageType?: 'home' | 'inner';
-  },
-  footerRef
-) {
+export const Layout = ({
+  children,
+  hasTOC = true,
+  pageDescription,
+  pageTitle,
+  pageType = 'inner',
+  platform,
+  url
+}: {
+  children: any;
+  hasTOC?: boolean;
+  pageDescription?: string;
+  pageTitle?: string;
+  pageType?: 'home' | 'inner';
+  platform?: Platform;
+  url?: string;
+}) => {
+  const [menuOpen, toggleMenuOpen] = useState(false);
+  const [tocHeadings, setTocHeadings] = useState<Heading[]>([]);
+
   useEffect(() => {
-    trackPageVisit();
-  }, []);
+    const headings: Heading[] = [];
+    const pageHeadings = document.querySelectorAll('.main > h2, .main > h3');
```

Comment:
Maybe we'll need to update this? But it seems like so far we don't want to show headings that are nested in other components (like accordions and tabs) and just want to show top level headings.